### ISSUE TITLE
`TCH004`: False positive caused by regression in v0.0.246

### ISSUE THREAD

**Issue by ngnpope** [charliermarsh/ruff#2852]
For the following code:

```python
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import Final, Literal, TypeAlias

    RatingKey: TypeAlias = Literal["good", "fair", "poor"]

RATING_KEYS: Final[tuple[RatingKey, ...]] = ("good", "fair", "poor")
```

I get the following:

```console
❯ pipx inject ruff ruff==0.0.245
  injected package ruff into venv ruff
done! ✨ 🌟 ✨

❯ ruff --version
ruff 0.0.245

❯ ruff --isolated --select TCH004 bug.py 

❯ pipx inject ruff ruff==0.0.246
  injected package ruff into venv ruff
done! ✨ 🌟 ✨

❯ ruff --version
ruff 0.0.246

❯ ruff --isolated --select TCH004 bug.py 
bug.py:6:31: TCH004 Move import `typing.Literal` out of type-checking block. Import is used for more than type hinting.
Found 1 error.
```

I suspect this is related to #2774 and #2779.

**Comment by charliermarsh** [comment]
Thank you, will fix.
